src/services/fetchAuth.js

```diff
@@ -0,0 +1,30 @@
+import axios from "axios";
+
+const fetchLoginDetails = async (userDetails, login = false) => {
+  const apiResponse = { token: "", username: "", error: "" };
+  try {
+    let response;
+    if (login) {
+      response = await axios.post("/api/auth/login", userDetails);
```

Comment:
make these seperate services for login and sign up as the code scales and you want to add emaill / otp verification it becomes difficult just a suggestion

Reply:
Thankyou for your suggestion. Have made the changes